llvm/include/llvm/Object/DXContainer.h

```diff
@@ -140,6 +159,22 @@ class PSVRuntimeInfo {
 
   const InfoStruct &getInfo() const { return BasicInfo; }
 
+  template <typename T> const T *getInfoAs() const {
+    if (const auto *P = std::get_if<dxbc::PSV::v2::RuntimeInfo>(&BasicInfo))
```

Comment:
Could we just return std::get_if<T>(&BasicInfo) for getInfoAs()?

Reply:
That wouldn’t do the same thing. This allows us to do `getInfoAs<dxbc::PSV::v1::RuntimeInfo>()` and get back a v1 pointer even when the underlying object is a `v2` object, or get a `v0` object regardless of the underlying type.

Reply:
Why do we want to get a v0::RuntimeInfo ptr from a v2 object?
Because v0::RuntimeInfo::swapBytes() is not included in v2::RuntimeInfo::swapBytes()?

Reply:
This is useful for cases where you want to read a field from the `v0` or `v1` data, but you don't actually care if the underlying data is a different version. You can see it in use in the `usesViewID()` method. That bit is part of the `v1` structure, so I want to return its value for `v1` or `v2` and return `false` for `v0`.

This helper allows me to not need to replicate the `std::get_if` blocks throughout the class, and since `std::is_same` will be compile-time resolved, the instantiations should be stripped down based on the statically resolvable structure versions.